compiler/src/iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -1045,6 +1047,63 @@ static unsigned decideFusableLinalgOps(FunctionOpInterface funcOp,
   return numRootOps;
 }
 
+FusionGroupMapping decideFusionGroups(FunctionOpInterface funcOp,
+                                      DominanceInfo const &dominanceInfo) {
+  unsigned numRoots = decideFusableLinalgOps(funcOp, dominanceInfo);
```

Comment:
Please add some TODOs on the usage of attributes here and below.
In a future PR we could refactor to return the roots without adding the attributes in the general case and feed that to a function that sets the attributes for IREE's current usage.

Reply:
If we are going down this path, could we just invest in a new op in Flow dialect, one that is not isolated from above, and just yields the results. So the `decideFusionGroups` could just create this new op. THe rest of the code in `DispatchLinalgOnTensors` is just related to make it isolated from above. So if the new op is `flow.dispatch.group` then you could have two paths
1) `flow.dispatch.group` -> `flow.dispatch.workgroups`
2) `flow.dispatch.group` -> `scf.for_each`.

But really long term we should not rely on `scf.for_each` construct. We have debated it a lot, and it doesnt gel well with IREEs constraints. It will be hard to maintain hooks into existing paths to support something that isnt clear is going to be adopted in IREE long-term.....